app/events/rtjc_test.go

```diff
@@ -34,34 +33,139 @@ func TestRtjc_isPinned(t *testing.T) {
+						return []string{"summary", "summary2"}, nil
+					}
+
+					// Regular link
+					return []string{"summary"}, nil
+				},
+			}
+
+			wasRead := false
+			io := &mocks.IoReader{
+				ReadFunc: func(p []byte) (n int, err error) {
+					if wasRead {
+						return 0, nil
+					}
+					output := []byte(tt.input + "\n")
+					copy(p, output)
+					wasRead = true
+					return len(output), nil
+				},
+			}
+
+			rtjc := makeRtjc(sb, sm)
+			rtjc.readMessage(context.Background(), io)
+			rtjc.Swg.Wait()
+
+			assert.Equal(t, tt.callsSubmit, len(sb.SubmitCalls()))
+			if tt.callsSubmit == 1 && len(sb.SubmitCalls()) == 1 {
+				assert.Contains(t, sb.SubmitCalls()[0].Text, tt.input)
+			}
+			assert.Equal(t, tt.callsSubmitHTML, len(sb.SubmitHTMLCalls()))
+			assert.Equal(t, tt.callsSummary, len(sm.GetSummariesByMessageCalls()))
+		})
 	}
+}
 
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		assert.Equal(t, http.MethodGet, r.Method)
-		assert.Equal(t, "token123", r.URL.Query().Get("token"))
-		_, err := w.Write([]byte(`{"content": "some content", "title": "some title"}`))
-		require.NoError(t, err)
-	}))
-
-	rtjc := Rtjc{OpenAISummary: oai, URClient: ts.Client(), UrAPI: ts.URL, UrToken: "token123"}
-
-	{
-		title, txt, err := rtjc.summary("some message blah https://example.com")
-		require.NoError(t, err)
-		assert.Equal(t, "ai summary", txt)
-		assert.Equal(t, "some title - some content", oai.SummaryCalls()[0].Text)
-		assert.Equal(t, "some title", title)
+func TestRtjc_SendSummary(t *testing.T) {
+	tbl := []struct {
+		name            string
+		input           string
+		summaryResponse []string
+		callsSubmit     int
+		callsSubmitHTML int
+		callsSummary    int
+	}{
+		{"Begin", "⚠️ Вещание подкаста началось - https://stream.radio-t.com/", []string{}, 0, 0, 1},
+		{"New theme", "⚠️ blah blah - https://link.example.com", []string{"summary"}, 0, 1, 1},
+		{"Remark", "⚠️ blah blah - https://radio-t.com/p/2023/04/04/prep-853/", []string{"summary1/2", "summary2/2"}, 0, 2, 1},
+		{"Remark with empty summaries", "⚠️ blah blah - https://radio-t.com/p/2023/04/04/prep-853/", []string{"summary1/3", "", "summary2/3", "summary3/3"}, 0, 3, 1},
+		{"Blah", "blah blah - https://link.example.com", []string{}, 0, 0, 0},
 	}
 
-	{
-		title, txt, err := rtjc.summary("some message blah https://radio-t.com")
-		require.NoError(t, err)
-		assert.Equal(t, "", txt)
-		assert.Equal(t, "", title)
+	for _, tt := range tbl {
+		t.Run(tt.name, func(t *testing.T) {
+			sb := &mocks.Submitter{
+				SubmitFunc: func(ctx context.Context, text string, pin bool) error {
+					return nil
+				},
+				SubmitHTMLFunc: func(ctx context.Context, text string, pin bool) error {
+					return nil
+				},
+			}
+
+			sm := &mocks.Summarizer{
+				GetSummariesByMessageFunc: func(link string) (messages []string, err error) {
+					return tt.summaryResponse, nil
+				},
+			}
+
+			rtjc := makeRtjc(sb, sm)
```

Comment:
to this this test doesn't seem realistic. In real life you don't create rtjc on ech send call

Reply:
I need to reset mocks. Unfortunately, `moq` doesn't support it.

And this test isn't sequences of calls. This is independent cases with same code of execution

Reply:
actually it does. run it ` -with-resets` and it will add reset methods to the generated code

Reply:
> And this test isn't sequences of calls. This is independent cases with same code of execution

Well, this is the hope and, most likely, the reality for now. However, someday I can see us introducing some state somewhere, and this artificially reset struct won't help in regression testing